### PR TITLE
[Vision] Improvements to documentation, inconsistent API, and fix legacy examples

### PR DESCRIPTION
**Pull request by zhreshold**
*Issue #, if available:*
Continue #733

*Description of changes:*

#### remaining unsolved comments from previous PR
it's still unclear what's the right choice or happens to be on the plan for migration towards v0.1

- Using `random_state` vs `random_seed` in `fit` function, I see both terms being heavily used in autogluon
- How to align with Tabular for `predict` and `predict_proba`
- `time_limit` vs `time_limits`
- `def predict(self, x)` or `def predict(self, dataset)`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: extra/setup.py [hunk 1]
```diff
@@ -46,9 +46,9 @@ def create_version_file():
 
 requirements = [
     'numpy>=1.16.0',
-    'gluoncv==0.9.0',
+    'gluoncv==0.10.0b20201216',
```
Comment (Innixma) [inline extra/setup.py:49]: Why is gluoncv present twice in this requirements list?

File: vision/src/autogluon/vision/image_prediction/image_prediction.py [hunk 2]
```diff
@@ -134,26 +136,27 @@ def fit(self,
             return
 
         # new HPO task
+        if time_limit is None and num_trials is None:
```
Comment (Innixma) [inline vision/src/autogluon/vision/image_prediction/image_prediction.py:139]: Is this check missing in object detection?

**Review by Innixma — APPROVED** [review]
Looks good! In regards to the remaining issues, we can plan on addressing them once the Tabular API refactor PR is merged in.

File: extra/setup.py (hunk 1 above)
Comment (Innixma) [reply extra/setup.py:49]: Also, how necessary is it to be using pre-releases from gluoncv? What if v0.10 isn't released by the time we want to release v0.1 AG?

**Comment by szha** [comment]
Job PR-828-5 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-828/5/index.html

**Pull request merged**